### PR TITLE
Wav encoder

### PR DESCRIPTION
**Pull request by breznak**
WAV encoder (using scypi) 

Dear guys, I'm offering this contribution in exchange for acceptance to this elite `Audio` club :wine_glass: :wink: 

Addresses #22

### CODE REVIEW

File: nupicaudio/encoders/iteration.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+class IterationEncoder(Encoder):
+  __metaclass__ = abc.ABCMeta
+  """ 
+  Iteration encoder encapsulates parsing of some data, and then creates an iterator over the
+  (otherwise monolithic) file. 
+
+  An example is a WAV file, during init() the file is transformed to a numeric representation (array), 
+  and each call of encode() moves over one element of this array. 
+
+  Typical applicatios for IterationEncoder: binary files (sound, video, biosignals, ...)
```
Comment (breznak) [inline nupicaudio/encoders/iteration.py]: this is not a 100% pure "encoder" functionality, but I like how it works for the binary files, esp. WAV,...

File: nupicaudio/encoders/sound_encoders_test.py [hunk 2]
```diff
@@ -0,0 +1,13 @@
+# This file will demanstrate use of available sound encoders:
+
+# WAV
+data="../examples/data/a0007.wav"
+#from nupicaudio.encoders.wave import WAVEncoder #FIXME move to nupicaudio.examples
+from wave import WAVEncoder
```
Comment (breznak) [inline nupicaudio/encoders/sound_encoders_test.py]: use the absolute import `from nupicaudio.encoders.wave` once `pip installable` #24

File: nupicaudio/encoders/wave.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
+# WAV encoder - use any available transformators
+
+from iteration import IterationEncoder
+
+from enum import Enum
+
+class WAVEncoder(IterationEncoder):
+
+  MODE = Enum('MODE', 'SCIPY') # TODO add others, matlab, ...
```
Comment (breznak) [inline nupicaudio/encoders/wave.py]: someone worked with MIDI files here, can you add/show me the code for parsing a midi, so we could integrate it in this encoder? (later, in a new PR)

**Comment by breznak** [comment]
This will be much more useful when #24 works!

**Comment by rhyolight** [comment]
@breznak I gave you write access, but @rcrowder is the de facto admin of this repo. Merge as you see fit, but I haven't heard from Richard in awhile.

**Comment by rcrowder** [comment]
@breznak With your write access you should now be able to merge changes. I'll try and keep a watch over any changes you make. Feel free to add to this repo as you see fit. :+1:

**Comment by passiweinberger** [comment]
Really cool one! 
What would you think about a midi encoder? @breznak

**Comment by breznak** [comment]
> ...MIDI encoder?

@passiweinberger I want to do it! see #22 (if you have a python parser, please show me how it's used, and we can make a PR - I think adding a new `MODE` to this `WAV` encoder would be OK (and we should call it `SoundEncoder` then)

**Comment by breznak** [comment]
FYI @rcrowder I merged this, as the WAVEncoder is OK, the issue with pip is separate (and easier to test now).

**Pull request merged**